### ISSUE TITLE
macOS no longer works

### ISSUE THREAD

**Issue by notrobpike** [nicollasricas/vscode-streamdeck#56]
macOS Sonoma 14.6.1 (latest) on intel (MBP 2019)
VS Code 1.92.2 (latest)
Stream Deck 6.7.1 (latest)
Stream Deck VSCode plugin 5.1.3 (latest?)
Stream Deck Firmware 1.03.000 (latest)

The plugin is reported as 5.1.3 in the stream deck prefs, however this is marked as the 4.1.5 release from github. I downloaded this long ago. Same failure mode with Stream Deck 6.6.0 and FW 1.02.005.

Stream Deck itself does work -- I can do Finder actions for example.

The status bar in VSCode reports `Deck | Connected`. Clicking it to make it active results in it changing to `Deck | Connecting` for 5s or so then it goes back to `Connected`. I am unable to discover where logs might be found to help diagnose this. This broke for me recently. If I had to guess it was with the 14.6.1 OS update.

On an M2 mac, same software versions, brand new stream deck and plugin install (it was never installed before), the status never gets past `Deck | Connecting`. I cannot `nc` to port 48969 either. So something is preventing the plugin to listen on 48969? Absolute default/fresh install with no prior settings present. On that machine, macOS firewall is off, but LittleSnitch is installed, in silent allow mode. I don't believe LittleSnitch has the ability to block localhost connections anyway. Also another oddity, the plugin claims version 5.1.2. This was a brand new download since I'd never had it installed before. I'm wondering why a brand new download reports as an older version than my years old plugin install.

**Comment by notrobpike** [comment]
I found the output panel from a similar issue. On the M2 mac it just showed what i was expecting -- inability to connect. Then I fired VSCode back up on the intel mac -- my primary use case -- and it now launches claiming `Active`.

However, it claims that for all windows. I believe it is documented, and my experience has been, only one window can be active. I normally have a remote-ssh session and a local session and have to activate one or the other. Fine I guess, but even though It claims `active` it doesn't actually work. I can also click to activate (even though already "active") and that seems to succeed but then it gets unexpected end of JSON input.

```
[8/29/2024, 3:13:54 PM] Message received, ActiveSessionChangedMessage.: {"id":"ActiveSessionChangedMessage","data":"{\"sessionId\":\"09f3f8ec-4229-42c0-a654-f7f9617ec8891724969428795\"}"}
[8/29/2024, 3:13:58 PM] Message received, ExecuteCommandMessage.: {"id":"ExecuteCommandMessage","data":"{\"command\":\"orkbench.action.showCommands\",\"arguments\":\"\"}"}
[ERROR] Unexpected end of JSON input, Stack: SyntaxError: Unexpected end of JSON input
	at JSON.parse (<anonymous>)
	at executeCommand (/Users/frank/.vscode/extensions/nicollasr.vscode-streamdeck-4.1.6/out/extension.js:99:37)
	at EventDispatcher.<anonymous> (/Users/frank/.vscode/extensions/nicollasr.vscode-streamdeck-4.1.6/out/extension.js:64:68)
	at Timeout._onTimeout (/Users/frank/.vscode/extensions/nicollasr.vscode-streamdeck-4.1.6/node_modules/ste-core/dist/events/Subscription.js:37:24)
	at listOnTimeout (node:internal/timers:573:17)
	at process.processTimers (node:internal/timers:514:7)
```

**Comment by notrobpike** [comment]
Looking at the error, I'm not sure if the "input" means in the direction of vscode->extension or from extension->vscode. I can see that the message "received" vscode->extension is well-formed JSON. *However* the command to execute has a typo `orkbench` vs `workbench`. I haven't edited the config in forever so I've no idea how this might have gotten introduced. This just happened to be the first button I tested on the Stream Deck, and the only one that looks like it contains a typo. After trying that button, all other buttons also failed, even though they have correct commands!

After fixing the typo, everything works again on the intel Mac. With the new behavior that I don't have to manually activate. The activation just follows whatever window is in the foreground.

On my M2, I continue to have the problem that 48969 is not listening. I tried rebooting and that did not fix it. AFAICT there's no conflicting service listening on 48969: `lsof -i :48969` returns nothing.
